resourcemanager/features/user_specified_segments.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package features
+
+// TreatUserSpecifiedSegmentsAsCaseInsensitive is a feature-toggle which specifies whether User Specified
+// Resource ID Segments should be compared case-insensitively as required.
+//
+// @tombuildsstuff: whilst this IS EXPOSED in the public interface - this is NOT READY FOR USE and should
+// not be exposed in user-focused logic at this time, else this'll become a source of knock-on problems
+// rather than being useful.
+//
+// There are a number of dependencies to enabling this, including completing the standardiation on the
+// `ResourceId` interface and the `ResourceIDReference` schema types - and surrounding updates.
+var TreatUserSpecifiedSegmentsAsCaseInsensitive = false
```

Comment:
Despite the comment I could still see this creeping into places where it shouldn't in the provider. Would it make sense and/or be possible for this to be accompanied by a check in the provider to flag new additions of this?

Reply:
By this, do you mean a check in the Provider to ensure that we're not doing `.ID() ==`? If so, yes - that can be added at the same time as fixing the 6 instances above

Reply:
I actually mean a check in the Provider to ensure we're not setting `features.TreatUserSpecifiedSegmentsAsCaseInsensitive = true` unless it's been explicitly added to a list of exceptions, like we do with [these checks](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/provider/provider.go)

Reply:
Yep, makes sense - we probably want both checks actually.

It's worth noting that since this feature-flag is for the behaviour as a whole (i.e. across the entire Provider) rather than being for individual Resource ID types - we'd only set this one-time rather than in multiple places anyway.

Reply:
For info: I've intentionally not included the validation changes in https://github.com/hashicorp/terraform-provider-azurerm/pull/25873 since this wants to be done in a follow up PR - will send that in the morning: